From 54f4edcbaccbc3bdf6ce262c53e625032b22d3e8 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sat, 18 Jan 2020 13:59:28 +0100
Subject: [PATCH] meson: add examples option

Add an option to allow the user to disable examples, this will allow to
build pipewire without alsa. Without this option, build with
-Dpipewire-alsa=false -Dalsa=false fails on:

src/examples/meson.build:47:0: ERROR: Unknown variable "alsa_dep".

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Signed-off-by: James Hilliard <james.hilliard1@gmail.com>
[Upstream status:
https://gitlab.freedesktop.org/pipewire/pipewire/merge_requests/217]
---
 meson_options.txt | 4 ++++
 spa/meson.build   | 4 +++-
 src/meson.build   | 4 +++-
 3 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/meson_options.txt b/meson_options.txt
index 5a738ddb..e8eece7f 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -2,6 +2,10 @@ option('docs',
        description: 'Build documentation',
        type: 'boolean',
        value: false)
+option('examples',
+       description: 'Build examples',
+       type: 'boolean',
+       value: true)
 option('man',
        description: 'Build manpages',
        type: 'boolean',
diff --git a/spa/meson.build b/spa/meson.build
index 42780a8c..37c034a4 100644
--- a/spa/meson.build
+++ b/spa/meson.build
@@ -39,7 +39,9 @@ endif
 
 subdir('tools')
 subdir('tests')
-subdir('examples')
+if get_option('examples')
+  subdir('examples')
+endif
 
 pkgconfig.generate(filebase : 'libspa-@0@'.format(spaversion),
   name : 'libspa',
diff --git a/src/meson.build b/src/meson.build
index 41f03ac6..1b43663a 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -4,7 +4,9 @@ subdir('extensions')
 subdir('daemon')
 subdir('tools')
 subdir('modules')
-subdir('examples')
+if get_option('examples')
+  subdir('examples')
+endif
 subdir('tests')
 
 if get_option('gstreamer')
-- 
2.20.1

